app/scripts/metamask-controller.js

```diff
@@ -3372,7 +3372,7 @@ export default class MetamaskController extends EventEmitter {
     if (sender.url) {
       const { hostname } = new URL(sender.url);
       // Check if new connection is blocked if phishing detection is on
-      const phishingTestResponse = this.phishingController.test(hostname);
+      const phishingTestResponse = await this.phishingController.test(hostname);
```

Comment:
Is it possible that this results in connection failures or dropped messages? That was the concern I raised in the controllers PR, that blocking here to download a new phishing config might cause problems.

The actual port should be constructed synchronously, as there are no `await`'s until after that step (in `connectExternal`). So we shouldn't see a timeout in establishing the connection itself, as far as I know.

But I'm less sure about the individual messages. Do they have a timeout?

Reply:
Downloading a new config might be slow. On the order of seconds, depending on where you're downloading it from and how fast your connection is.

Reply:
@Gudahtt I've tried to simulate a very slow connection here:

https://user-images.githubusercontent.com/34557516/195891473-afb65c9b-5634-4b7d-a1a0-1f6381e4ff21.mov

I've commented out the if conditional to only update phishing lists if they're out of date, so that we await them every time.

With my as slow as it will let me go throttling profile it takes 2 seconds for the configs to load and I'm not seeing issues with dropped messages. 

Does this achieve the level testing confirmation that you were hoping to see?

Reply:
Yes, I think so, thanks